### DISCUSSION TITLE
What exactly Busy, Sched and Idle columns mean?

### DISCUSSION THREAD

**Discussion by link2xt** [tokio-rs/console#426] [Q&A]
I have looked at the https://docs.rs/tokio-console/latest/tokio_console/ documentation, but did not find the answer. How Busy, Sched and Idle columns are measured? Does "Total" column equal the sum of Busy, Sched and Idle? Why some tasks have Total = Sched = Busy and zero idle time like in the screenshot at https://github.com/deltachat/deltachat-core-rust/pull/4418?

**Comment by hds** [comment]
Total is the sum of Busy, Scheduled, and Idle.

* Busy is the time a task is being polled (this is when the task is "active" and doing work)
* Scheduled is the time that a task is ready to be polled, but hasn't yet been polled.
* Idle is the time after a task yields (finishes doing work) and is waiting on some resource (like a timer, a lock, or some other async resource).

I wrote a bit more about it here: https://hegdenu.net/posts/task-scheduled-time-in-console/

Let me know if that helps.

**Comment by link2xt** [comment]
> Total is the sum of Busy, Scheduled, and Idle.

Then why on the screenshot at the top of https://github.com/deltachat/deltachat-core-rust/pull/4418 there are tasks which have 38s Total, 38s Busy and 38s Sched time?

**Reply by hds** [reply]
This looks like a bug to me. It shouldn't be the case that Busy + Sched + Idle is more than Total.

Do you see the same in the detail view for that task?

If you could provide a minimal example for this issue, that would be fantastic. Otherwise, if it always occurs, maybe I can try to repro with deltachat myself.
